src/material/core/option/option.scss

```diff
@@ -32,8 +32,19 @@
   &.mdc-list-item--disabled {
     // This is the same as `mdc-list-mixins.list-disabled-opacity` which
     // we can't use directly, because it comes with some selectors.
-    opacity: mdc-list-variables.$content-disabled-opacity;
     cursor: default;
+
+    // Reduce opacity of the contents of a disabled item to give a "grayed out" appearance. Avoid
+    // changing opacity of the option element itself to support focusing on disabled options using
+    // keyboard.
+    > * {
+      opacity: mdc-list-variables.$content-disabled-opacity;
+    }
+
+    // Prevent clicking on disabled options with mouse. Support focusing on disabled option using
+    // keyboard, but not with mouse.
+    pointer-events: none;
```

Comment:
nit: `pointer-event: none` can be annoying, because it also prevents users from selecting the option's text. We could solve this by checking if the option is disabled in its `click` handler.

Reply:
Hi Kristiyan, unless I'm missing something. I don't think we can prevent users from clicking on an option *and* allow users to select option text. Here's why.

I tried calling `event.preventDefault()` and `event.stopPropagation()` in the click handlers. But that left a wierd issue where the disabled option would still get focus when clicked. Here's a video recording of the behavior I get.

https://user-images.githubusercontent.com/7720245/224885222-1e747746-8369-4e2a-95d8-511243e56ff1.mov

I could one way to prevent clicking on options from focusing them. I called `event.stopPropagation` on `mousedown`. Here's a draft PR that shows that implementaiton https://github.com/angular/components/pull/26783. Although this prevents clicking from focusing the option, I cannot select the option's text (just like when applying `pointer-events: none`. This is why I don't think there's a way to prevent mouse from focusing, but also allow selecting the option's text.

WDYT? Am I misunderstanding something, or do you think `pointer-events: none` is the right approach?